### DISCUSSION TITLE
Questions about Gauge labels

### DISCUSSION THREAD

**Discussion by 1138559250** [prometheus/client_python#896] [Q&A]
I'm a newbie with Prometheus_client.
I would like to ask if it is possible to customize the label.
The example is as follows
`metrics_one = prometheus_client.Gauge("mysql_data", "mysql_data", ["label1", "label2", "label3", "label4", "label5", "label6"])`
`metrics_two = prometheus_client.Gauge("mysql_data", "mysql_data", ["label1", "label2", ])`
`metrics_three = prometheus_client.Gauge("mysql_data", "mysql_data", ["label1", "label3", ])`
`metrics_four = prometheus_client.Gauge("mysql_data", "mysql_data", ["label1", "label2", "label4" ])`

But I have a lot of labels, I only need to use a few of them when I insert data，
Do I have to write all the metrics like this? Is there another way to do it?
And in the end, I finally hope to use flask to respond to the latest data of all labels of metrics.
`@app.route('/metrics')`
`def metrics():`
`    set_data_to_prometheus()`
`    return Response(prometheus_client.generate_latest(metrics), mimetype="text/plain")`

If so, I can only pass out the data of one metric.
Looking forward for your reply, thank you

**Comment by csmarchbanks** [comment]
Hello,

You could probably define all of the labels on only one metric, and then set them to `""` for cases where that label name does not have a value. In Prometheus an empty string is the same as a label not existing. Do you have a bit more of a concrete example for what the different labels would be? I could help point you in the proper direction if so, having a lot of empty labels on the same metric is not a pattern commonly seen.
